pkg/api/bzzfile_test.go

```diff
@@ -0,0 +1,107 @@
+// Copyright 2020 The Swarm Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package api_test
+
+import (
+	"bytes"
+	"fmt"
+	"io/ioutil"
+	"mime"
+	"net/http"
+	"testing"
+
+	"github.com/ethersphere/bee/pkg/api"
+	"github.com/ethersphere/bee/pkg/jsonhttp"
+	"github.com/ethersphere/bee/pkg/jsonhttp/jsonhttptest"
+	"github.com/ethersphere/bee/pkg/logging"
+	"github.com/ethersphere/bee/pkg/storage/mock"
+	"github.com/ethersphere/bee/pkg/swarm"
+	"github.com/ethersphere/bee/pkg/tags"
+)
+
+func TestBzz(t *testing.T) {
+	var (
+		simpleResource  = func() string { return "/bzz-file" }
+		addressResource = func(addr string) string { return "/bzz-file/" + addr }
+		simpleData      = []byte("this is a simple text")
+		mockStorer      = mock.NewStorer()
+		client          = newTestServer(t, testServerOptions{
+			Storer: mockStorer,
+			Tags:   tags.NewTags(),
+			Logger: logging.New(ioutil.Discard, 5),
+		})
+	)
+	t.Run("simple-upload", func(t *testing.T) {
+		jsonhttptest.ResponseDirectSendHeadersAndReceiveHeaders(t, client, http.MethodPost, simpleResource(), bytes.NewReader(simpleData), http.StatusBadRequest, jsonhttp.StatusResponse{
+			Message: "not a mutlipart/form-data",
+			Code:    http.StatusBadRequest,
+		}, nil)
+	})
+
+	t.Run("simple-upload", func(t *testing.T) {
+		fileName := "simple_file.txt"
+		rootHash := "295673cf7aa55d119dd6f82528c91d45b53dd63dc2e4ca4abf4ed8b3a0788085"
+		_ = jsonhttptest.ResponseDirectWithMultiPart(t, client, http.MethodPost, simpleResource(), fileName, simpleData, http.StatusOK, api.FileUploadResponse{
+			Address: swarm.MustParseHexAddress(rootHash),
+		})
+	})
+
+	t.Run("check-content-type-detection", func(t *testing.T) {
+		fileName := "my-pictures.jpeg"
+		rootHash := "602a37afed59ddc84746e17f090474502d24e4a6125714918c19e771c40b4477"
+		_ = jsonhttptest.ResponseDirectWithMultiPart(t, client, http.MethodPost, simpleResource(), fileName, simpleData, http.StatusOK, api.FileUploadResponse{
+			Address: swarm.MustParseHexAddress(rootHash),
+		})
+
+		rcvdHeader := jsonhttptest.ResponseDirectCheckBinaryResponse(t, client, http.MethodGet, addressResource(rootHash), nil, http.StatusOK, simpleData, nil)
+		cd := rcvdHeader.Get("Content-Disposition")
+		_, params, err := mime.ParseMediaType(cd)
+		if err != nil {
+			t.Fatal(err)
+		}
+		if params["filename"] != fileName {
+			t.Fatal("Invalid file name detected")
+		}
+	})
+
+	t.Run("upload-then-download-and-check-data", func(t *testing.T) {
+		fileName := "vacation.mp4"
```

Comment:
mp4 but html content? I don't understand..

Reply:
well.. the detection is based on content.. filename makes no difference.. for easy reading i am making it sample.html :-)